src/main/java/ch/njol/skript/lang/SyntaxElement.java

```diff
@@ -47,4 +50,22 @@ default ParserInstance getParser() {
 		return ParserInstance.get();
 	}
 
+	/**
+	 * Gets the events in which this element can be used in. If there are no limits to this element, you may
+	 * return an empty array, or simply not override this method.
+	 * @return the events this element is usable in, or an empty array.
+	 */
+	default Class<? extends Event>[] getUsableEvents() {
```

Comment:
I'm unsure about this one. I personally think we need to avoid adding more Bukkit stuff into Skript's internals. Update for discussion though :)

Reply:
I feel like as we move out of bukkit, this can still be around, but just as a usable executable, so a SkriptEvent class if that makes sense